### PR TITLE
Add stable staging environment inputs

### PR DESCRIPTION
**Pull request by mishapos**

### CODE REVIEW

File: action.yml [hunk 1]
```diff
@@ -42,7 +42,13 @@ inputs:
     required: false
   skip_api_build:
     description: "Skips the build step for the Azure Functions source code if set to true."
+  production_branch:
+    description: "Production branch. Deploys from other branches will be considered staging environments."
```
Comment (annikel) [inline action.yml]: Is it Deploys or Deployments?

**Review by annikel — APPROVED** [review]

File: action.yml (hunk 1 above)
Comment (mishapos) [reply action.yml]: `Deployments` is better. Thank you!

File: action.yml [hunk 2]
```diff
@@ -42,7 +42,13 @@ inputs:
     required: false
   skip_api_build:
     description: "Skips the build step for the Azure Functions source code if set to true."
```
Comment (joslinmicrosoft) [inline action.yml:44]: let's add back the `required: false` for this one

File: action.yml [hunk 3]
```diff
@@ -42,7 +42,13 @@ inputs:
...
+  production_branch:
+    description: "Production branch. Deployments from other branches will be considered preview environments."
     required: false
+  deployment_environment:
+    description: "Explicit name of Azure Static Web Apps environment for this deployment."
+    required: false
+
```
Comment (joslinmicrosoft) [inline action.yml]: nit: remove this newline

**Review by joslinmicrosoft — CHANGES_REQUESTED** [review]

File: action.yml (hunk 2 above)
Comment (mishapos) [reply action.yml:44]: Added it back. thanks for the good catch!

File: action.yml [hunk 4]
```diff
@@ -43,6 +43,12 @@ inputs:
   skip_api_build:
     description: "Skips the build step for the Azure Functions source code if set to true."
     required: false
+  production_branch:
+    description: "Production branch. Deployments from other branches will be considered preview environments."
```
Comment (miwebst) [inline action.yml]: Lets add "When specified," before deployments

File: action.yml [hunk 5]
```diff
@@ -43,6 +43,12 @@ inputs:
...
+    required: false
+  deployment_environment:
+    description: "Explicit name of Azure Static Web Apps environment for this deployment."
```
Comment (miwebst) [inline action.yml]: I feel like this should be something like "Name of the Azure Static Web Apps environment to deploy to"

**Review by miwebst — APPROVED** [review]

**Review by joslinmicrosoft — APPROVED** [review]

**Pull request merged**